usr/share/rear/lib/framework-functions.sh

```diff
@@ -122,7 +122,7 @@ function SourceStage () {
               "$BACKUP"/{default,"$ARCH","$OS","$OS_MASTER_VENDOR","$OS_MASTER_VENDOR_ARCH","$OS_MASTER_VENDOR_VERSION","$OS_VENDOR","$OS_VENDOR_ARCH","$OS_VENDOR_VERSION"}/*.sh \
               "$OUTPUT"/{default,"$ARCH","$OS","$OS_MASTER_VENDOR","$OS_MASTER_VENDOR_ARCH","$OS_MASTER_VENDOR_VERSION","$OS_VENDOR","$OS_VENDOR_ARCH","$OS_VENDOR_VERSION"}/*.sh \
     "$OUTPUT"/"$BACKUP"/{default,"$ARCH","$OS","$OS_MASTER_VENDOR","$OS_MASTER_VENDOR_ARCH","$OS_MASTER_VENDOR_VERSION","$OS_VENDOR","$OS_VENDOR_ARCH","$OS_VENDOR_VERSION"}/*.sh \
-                 | sed -e 's#/\([0-9][0-9][0-9]\)_#/!\1!_#g' | sort -t \! -k 2 | tr -d \! ) )
+                 | sed -e 's#/\([0-9][0-9][0-9]\)_#/!\1!_#g' | sort -t \! -k 2 -u | tr -d \! ) )
```

Comment:
@lzaoral 
could you explain or point to an URL that explains
the reason behind why there is 'sort ... -u'?
Sorry if stupid question but I am basically lost
with all those things here.
I assume 'sort ... -u' avoids that scripts get sourced more than once
because with '-u' sort outputs only the first one
so duplicates get discarded
and because of the sorting by leading script numbers
the scripts get run each one exactly once
and in the intended ordering.

Reply:
Does the commit message of https://github.com/rear/rear/pull/3171/commits/97c6b9de6904c343e0a1c6b9ea92e7958ec4720c answer it?

Reply:
Perhaps you could add a comment in the code?

Reply:
Plus that running a script within the intended ordering
more than once never makes sense.

Reply:
Ah - now even I finally see it in the commit message,
that it is even about running whole stages more than once
which even more than never makes sense ;-)

Reply:
@jsmeix Thank you for the feedback! :)  I'll reword the commit message and add some additional comments tomorrow.  And your last comment actually made me realise that I've used stage instead of a script. (Though, both should be executed at most once.)

Reply:
But now I see that in the code a bit above there is
```
local scripts=( $( cd $SHARE_DIR/$stage
```
and the whole function 'SourceStage' should source
only scripts within one single stage.

Either all that actually is totally confusing
or I am just too stupid to understand it.

@pcahyna @lzaoral 
don't worry - I give up trying to understand it.

@pcahyna 
better "just merge" it as soon as possible
before it endlessly runs in circles ;-)

Reply:
@lzaoral 
aaaahhhhh!
I noticed your above comment right now.
What a relief!
I am not mad.
:-)

Reply:
@lzaoral @pcahyna @rear/contributors 
I wish you all a relaxed evening and a recovering sleep tonight!

Reply:
@jsmeix I've reworded the commit message and added an additional comment in 0468a6a1efa467f84422f89143c70b65e2d64eee.  Please, let me know if it is now easier to understand or if I should make additional edits.